File: content/problems/0257-binary-tree-paths/solutions/dfs-path-string/solution.cpp
class Solution {
    void dfs(TreeNode* node,string path,vector<string>& answer){
        if(!node)return;
        string current=path.empty()?to_string(node->val):path+"->"+to_string(node->val);
        if(!node->left&&!node->right)answer.push_back(current);
        else{
            dfs(node->left,current,answer);
            dfs(node->right,current,answer);
        }
    }
    public:vector<string> binaryTreePaths(TreeNode* root){
        vector<string> answer;
        dfs(root,"",answer);
        return answer;
    }
};

File: content/problems/0257-binary-tree-paths/solutions/dfs-path-string/Solution.kt
class Solution {
    fun binaryTreePaths(root:TreeNode?):List<String>{
        val answer=mutableListOf<String>();
        fun dfs(node:TreeNode?,path:String){
            if(node==null)return;
            val current=if(path.isEmpty())node.`val`.toString() else path+"->"+node.`val`;
            if(node.left==null&&node.right==null)answer.add(current)else{
                dfs(node.left,current);
                dfs(node.right,current)
            }
        };
        dfs(root,"");
        return answer
    }
}
